k8s_snapshots/backends/google.py

```diff
@@ -206,19 +234,31 @@ def create_snapshot(
     # describes the input parameters, but contains output-only parameters,
     # the correct table can be found at
     # https://cloud.google.com/compute/docs/reference/latest/disks/createSnapshot#response
-    operation = gcloud.disks().createSnapshot(
-        disk=disk.name,
-        project=get_project_id(ctx),
-        zone=disk.zone,
-        body=request_body
-    ).execute()
-
-    return {
-        'snapshot_name': snapshot_name,
-        'zone': disk.zone,
-        'operation_name': operation['name']
-    }
-
+    if disk.regional:
```

Comment:
looks like there's a lot of duplicated code inside this conditinoal. how about you assign `gcloud.regionDisks()` or `gcloud.disks()` to a temp variable (`disks` or something?), and then run `createSnapshot()` on it outside?

Reply:
Issue here is that each of the functions require different parameters depending on which PVC type it is:

        ```operation = gcloud.regionDisks().createSnapshot(
￼            disk=disk.name,
￼            project=get_project_id(ctx),
￼            region=disk.region,
￼            body=request_body
￼        ).execute()```

Note that `region` is being set here.

```        operation = gcloud.disks().createSnapshot(
￼            disk=disk.name,
￼            project=get_project_id(ctx),
￼            zone=disk.zone,
￼            body=request_body
￼        ).execute()```

And here zone is being set. If you have other recommendations I'm all ears!

Reply:
oh i missed that. my bad!